--- a/src/integration/test_suites/smoke_test_hw_config/caliptra_isr.h
+++ b/src/integration/test_suites/smoke_test_hw_config/caliptra_isr.h
@@ -126,7 +126,7 @@
     }
 }
 
-inline void service_soc_ifc_notif_intr () {
+static inline void service_soc_ifc_notif_intr () {
     uint32_t * reg = (uint32_t *) (CLP_SOC_IFC_REG_INTR_BLOCK_RF_NOTIF_INTERNAL_INTR_R);
     uint32_t sts = *reg;
     /* Write 1 to Clear the pending interrupt */
