--- a/hw/ip/otp_ctrl/data/otp_ctrl_smoke_vseq.sv.tpl
+++ b/hw/ip/otp_ctrl/data/otp_ctrl_smoke_vseq.sv.tpl
@@ -13,6 +13,8 @@
   `uvm_object_utils(otp_ctrl_smoke_vseq)
 
   `uvm_object_new
+
+  localparam bit [TL_DW-1:0] TimeoutMax = '1;
 
   rand bit                           do_req_keys, do_lc_trans;
   rand bit                           access_locked_parts;
@@ -64,7 +66,7 @@
   }
 
   constraint check_timeout_val_c {
-    check_timeout_val inside {0, [100_000:'1]};
+    check_timeout_val inside {0, [100_000:TimeoutMax]};
   }
 
   constraint ecc_otp_err_c {ecc_otp_err == OtpNoEccErr;}
