--- a/hw/ip/spi_device/pre_dv/tb/spid_common.sv
+++ b/hw/ip/spi_device/pre_dv/tb/spid_common.sv
@@ -97,6 +97,7 @@
       payload_en:       4'b 0001, // MISO
       payload_dir:      PayloadIn,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
 
@@ -113,6 +114,7 @@
       payload_en:       4'b 0001, // MISO
       payload_dir:      PayloadIn,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
 
@@ -129,6 +131,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -145,6 +148,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -161,6 +165,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -177,6 +182,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -193,6 +199,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -209,6 +216,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -225,6 +233,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -241,6 +250,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -257,6 +267,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -273,6 +284,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -289,6 +301,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -305,6 +318,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -321,6 +335,7 @@
       payload_en:       4'b 0001, // MISO
       payload_dir:      PayloadIn,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 1,
       busy:             1'b 1
     },
@@ -337,6 +352,7 @@
       payload_en:       4'b 0000, // MISO
       payload_dir:      PayloadIn,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 1,
       busy:             1'b 1
     },
@@ -353,6 +369,7 @@
       payload_en:       4'b 0000, // MISO
       payload_dir:      PayloadIn,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 1,
       busy:             1'b 1
     },
@@ -369,6 +386,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -385,6 +403,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -401,6 +420,7 @@
       payload_en:       4'b 1111, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -417,6 +437,7 @@
       payload_en:       4'b 0011, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -433,6 +454,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -449,6 +471,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -465,6 +488,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -481,6 +505,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -497,6 +522,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -513,6 +539,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     },
@@ -529,6 +556,7 @@
       payload_en:       4'b 0010, // MISO
       payload_dir:      PayloadOut,
       payload_swap_en:  1'b 0,
+      read_pipeline_mode: RdPipeZeroStages,
       upload:           1'b 0,
       busy:             1'b 0
     }
